Solutions/GoogleCloudPlatformIAM/Analytic Rules/GCPIAMEmptyUA.yaml

```diff
@@ -0,0 +1,27 @@
+id: 9e0d8632-d33d-4075-979e-c972674f77b3
+name: GCP IAM - Empty user agent
+description: |
+  'Detects requests where user agent is empty.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: GCPIAMDataConnector
+    dataTypes:
+      - GCP_IAM
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - DefenseEvasion
+relevantTechniques:
+  - T1550
+query: |
+  GCP_IAM
+  | where isempty(HttpUserAgentOriginal)
+  | extend AccountCustomEntity = PayloadAuthenticationinfoPrincipalemail

```

Comment:
Wondering if there is an IPCustomEntity as well that could be added. Can we also have the timestamp field mapped please  (i.e. extend timestamp = TimeGenerated). Thanks